blaze/expr/datetime.py

```diff
@@ -222,10 +246,149 @@ def truncate(expr, *args, **kwargs):
     return DateTimeTruncate(expr, measure, normalize_time_unit(unit))
 
 
+class UnaryDateTimeFunction(ElemWise):
+
+    """DateTime function that only takes a single argument."""
+    _arguments = '_child'
+
+
+class Round(ElemWise):
+    _arguments = '_child', 'freq'
+
+    @property
+    def schema(self):
+        return self._child.schema
+
+
+class Floor(ElemWise):
+    _arguments = '_child', 'freq'
+
+    @property
+    def schema(self):
+        return self._child.schema
+
+
+class Ceil(ElemWise):
+    _arguments = '_child', 'freq'
+
+    @property
+    def schema(self):
+        return self._child.schema
+
+
+class dt_ns(object):
+
+    def __init__(self, field):
+        self.field = field
+
+    def year(self):
+        return year(self.field)
+    def month(self):
+        return month(self.field)
+    def day(self):
+        return day(self.field)
+    def hour(self):
+        return hour(self.field)
+    def minute(self):
+        return minute(self.field)
+    def date(self):
+        return date(self.field)
+    def time(self):
+        return time(self.field)
+    def second(self):
+        return second(self.field)
+    def millisecond(self):
+        return millisecond(self.field)
+    def microsecond(self):
+        return microsecond(self.field)
+    def nanosecond(self):
+        return nanosecond(self.field)
+    def weekday(self):
+        return weekday(self.field)
+    def weekday_name(self):
+        return weekday_name(self.field)
+    def daysinmonth(self):
+        return daysinmonth(self.field)
+    def weekofyear(self):
+        return weekofyear(self.field)
+    def dayofyear(self):
+        return dayofyear(self.field)
+    def dayofweek(self):
+        return dayofweek(self.field)
+    def quarter(self):
+        return quarter(self.field)
+    def is_month_start(self):
+        return is_month_start(self.field)
+    def is_month_end(self):
+        return is_month_end(self.field)
+    def is_quarter_start(self):
+        return is_quarter_start(self.field)
+    def is_quarter_end(self):
+        return is_quarter_end(self.field)
+    def is_year_start(self):
+        return is_year_start(self.field)
+    def is_year_end(self):
+        return is_year_end(self.field)
+    def days_in_month(self):
+        return days_in_month(self.field)
+    def strftime(self, format):
```

Comment:
For the methods that take a `format` string argument, when is validation done?  At a minimum we want to raise a `TypeError` as early as possible if `format` isn't a string; ideally we'd validate that `format`'s value is correct, if possible.